lib/gears/surface.lua

```diff
@@ -205,6 +206,55 @@ function surface.apply_shape_bounding(draw, shape, ...)
   draw.shape_bounding = img._native
 end
 
+local function run_in_hierarchy(self, cr, width, height)
+    local function main_layout(h)
+        h:update({dpi=96}, self, width, height)
```

Comment:
Are you sure? I am just asking, but isn't `widget::layout_changed` supposed to end up calling this? If I create an `.svg` at main loop cycle `#1`, then do `:finish()` at cycle `#10` after chaning the widget 9 time, isn't it better to have this called? If not, then I will remove it, test and merge this PR as it is the only remaining issue.